sql/core/src/main/scala/org/apache/spark/sql/jdbc/README.md

```diff
@@ -0,0 +1,81 @@
+---
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+# JDBC Connection Provider Handling In Spark
+
+This document aims to explain and demystify JDBC connection providers as they are used by Spark
+and the usage or custom provider implementation is not obvious.
+
+## What are JDBC connection providers and why use them?
+
+JDBC connection providers (CPs from now on) are making JDBC connections initiated by JDBC sources.
+When a Spark source initiates JDBC connection it looks for a CP which supports the included driver,
+the user just need to provide the `keytab` location and the `principal`. The `keytab` file must exist
+on each node where connection is initiated. The way how CP lookup happens described in later chapter.
+
+Spark initially provided non-authenticated or user/password authenticated connections.
+This is quite insecure and some of the users expected stronger authentication possibilities.
+This need fulfilled in 2 ways:
+ * Embedded CPs added which support kerberos authentication using `keytab` and `principal` (but only if the JDBC driver supports keytab)
```

Comment:
Let's use either "built-in" or "embedded" and try to be consistent with sql jdbc doc.

Reply:
Switched to built-in.